partiql-lang/src/main/kotlin/org/partiql/lang/planner/PartiQLPhysicalPass.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at:
+ *
+ *       http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
+package org.partiql.lang.planner
+
+import org.partiql.lang.domains.PartiqlPhysical
+
+/**
+ * Defines a [PartiQLPlannerPass] of a [PartiqlPhysical.Plan]
+ */
+fun interface PartiQLPhysicalPass : PartiQLPlannerPass<PartiqlPhysical.Plan>
```

Comment:
I know this is similar to today, but I believe we be focusing on making the logical domain the public interface for planning and rewriting as the logical plan is what we transport. The physical plan stuff is really Kotlin internal only.

Similar comment from substrait
> There is no true distinction between logical and physical operations in Substrait. By convention, certain operations are classified as physical, but all operations can be potentially used in any kind of plan. A particular set of transformations or target operators may (by convention) be considered the “physical plan” but this is a characteristic of the system consuming substrait as opposed to a definition within Substrait.

That is, the physical part is system specific, but what is defined by the PartiQL Canonical Plan (ie public interface) is not physical

Reply:
But because the "impl" field is so minimal, I'm beginning to think each we should allow annotating logical plan nodes

Reply:
I'd personally recommend implementing this in a separate PR. This code linked doesn't change any API. I just moved this top-level interface from PartiQLPlannerPass.kt to conform with one top-level interface/class per file.